lib/ace/ext/papeeria/katex-previewer.coffee

```diff
@@ -56,26 +61,62 @@ define((require, exports, module) ->
       return
 
     jqEditorContainer = $(editor.container)
-    jqFormula = -> $("#formula")
+    getFormulaElement = -> $("#formula")
 
     ch = contextHandler = {
       contextPreviewExists: false
       updateDelay: 1000
 
-      removeRegex: /\\end\{equation\}|\\begin\{equation\}|\\label\{[^\}]*\}/g
-
       getEquationRange: (cursorRow) ->
         i = cursorRow
-        while latexContextParser.getContext(editor.session, i - 1) == "equation"
+        while latexContextParser.getContext(editor.getSession(), i - 1) == "equation"
           i -= 1
         start = i
-        while latexContextParser.getContext(editor.session, i + 1) == "equation"
+        while latexContextParser.getContext(editor.getSession(), i + 1) == "equation"
           i += 1
         end = i
         return [start, end]
 
+      compareTokens: (tokenToCheck, tokenToMatch) ->
+        return tokenToCheck["type"] == tokenToMatch["type"] and (new RegExp(tokenToMatch["value"])).test(tokenToCheck["value"])
+
+      filterTokens: (tokens) ->
```

Comment:
Removing a sequence is not the same as removing elements of the sequence one by one?